SingularityUI/app/components/newDeployForm/NewDeployForm.jsx

```diff
@@ -0,0 +1,1303 @@
+    {id: 'serviceBasePath', type: 'text'},
+    {id: 'loadBalancerGroups', type: 'array', arrayType: 'text'},
+    {id: 'loadBalancerOptions', type: 'map'},
+    {id: 'loadBalancerPortIndex', type: 'text', default: 0},
+    {
+      id: 'containerInfo',
+      type: 'object',
+      values: [
+        {id: 'type', type: 'text', default: 'mesos'},
+        {
+          id: 'docker',
+          type: 'object',
+          values: [
+            {id: 'image', type: 'text'},
+            {id: 'network', type: 'text', default: 'NONE'},
+            {id: 'parameters', type: 'map'},
+            {id: 'privileged', type: 'text'},
+            {id: 'forcePullImage', type: 'text'},
+            {id: 'volumes', type: 'array', arrayType: 'volume'},
+            {id: 'portMappings', type: 'array', arrayType: 'portMapping'}
+          ]
+        }
+      ]
+    },
+    {
+      id: 'resources',
+      type: 'object',
+      values: [
+        {id: 'cpus', type: 'number', default: 1},
+        {id: 'memoryMb', type: 'number', default: 128},
+        {id: 'numPorts', type: 'number', default: 0}
+      ]
+    }
+  ],
+  defaultExecutor: [
+    {id: 'command', type: 'text'},
+    {id: 'uris', type: 'array', arrayType: 'text'},
+    {id: 'arguments', type: 'array', arrayType: 'text'}
+  ],
+  customExecutor: [
+    {id: 'customExecutorCmd', type: 'text'},
+    {
+      id: 'executorData',
+      type: 'object',
+      values: [
+        {id: 'cmd', type: 'text'},
+        {id: 'extraCmdLineArgs', type: 'array', arrayType: 'text'},
+        {id: 'user', type: 'text', default: 'root'},
+        {id: 'sigKillProcessesAfterMillis', type: 'number', default: 120000},
+        {id: 'successfulExitCodes', type: 'array', arrayType: 'number'},
+        {id: 'maxTaskThreads', type: 'text'},
+        {id: 'loggingTag', type: 'text'},
+        {id: 'loggingExtraFields', type: 'map'},
+        {id: 'preserveTaskSandboxAfterFinish', type: 'text'},
+        {id: 'skipLogrotateAndCompress', type: 'text'},
+        {id: 'embeddedArtifacts', type: 'array', arrayType: 'artifact'},
+        {id: 'externalArtifacts', type: 'array', arrayType: 'artifact'},
+        {id: 's3Artifacts', type: 'array', arrayType: 'artifact'}
+      ]
+    }
+  ]
+};
+
+function makeFlattenedFields(fields) {
+  const flattenedFields = {};
+  for (const field of fields) {
+    if (field.type === 'object') {
+      _.extend(flattenedFields, makeFlattenedFields(field.values));
+    } else {
+      flattenedFields[field.id] = field;
+    }
+  }
+  return flattenedFields;
+}
+
+const FLATTENED_FIELDS = _.extend(
+  {},
+  makeFlattenedFields(FIELDS.all),
+  makeFlattenedFields(FIELDS.customExecutor),
+  makeFlattenedFields(FIELDS.defaultExecutor)
+);
+
+class NewDeployForm extends Component {
+
+  componentDidMount() {
+    this.props.clearForm(FORM_ID);
+  }
+
+  updateField(fieldId, newValue) {
+    this.props.update(FORM_ID, fieldId, newValue);
+  }
+
+  getValue(fieldId) {
```

Comment:
Is it worth having this?

I don't see any prop validation here, so I don't know if form is a required prop, but I think it should be. Then this field can be removed.